presto-main/src/main/java/io/prestosql/memory/QueryContext.java

```diff
@@ -110,13 +114,26 @@ public QueryContext(
                 newRootAggregatedMemoryContext(new QueryMemoryReservationHandler(this::updateSystemMemory, this::tryReserveMemoryNotSupported), 0L));
     }
 
+    public boolean needsMemoryLimitsInitialized()
+    {
+        return needsMemoryLimitsInitialized;
+    }
+
+    public synchronized void setMemoryLimitsInitialized()
+    {
+        needsMemoryLimitsInitialized = false;
+    }
+
     // TODO: This method should be removed, and the correct limit set in the constructor. However, due to the way QueryContext is constructed the memory limit is not known in advance
     public synchronized void setResourceOvercommit()
     {
+        isResourceOverCommit = true;
         // Allow the query to use the entire pool. This way the worker will kill the query, if it uses the entire local general pool.
```

Comment:
Before this change it wasn't necessarily broken, since the subsequent task update would trigger the memory limit reset based on the current pool max size. This change is required in order to preserve that behavior (although it performs it earlier) so I'm not sure it's really something that should be split into its own commit.

Reply:
@pettyjamesm yes the behavior wasn't broken, I just meant to update the comment on line 131 to reflect the behavior correctly. The worker kills the query if the corresponding pool limit is reached, as opposed to the "general" pool as mentioned in the comment. (It's an independent thing, but doesn't matter much if you'd like to do it in the same commit.)

Reply:
Ah, ok- I misunderstood. Added a new commit to clear up that comment and fix one other javadoc reference.